### PR TITLE
Added exception handling for date parsing and storage and refactoring of taskManager methods

### PR DESCRIPTION
**Pull request by kyhjonathan**

### CODE REVIEW

File: src/main/java/data/Event.java [hunk 1]
```diff
@@ -92,5 +99,32 @@ public String getSaveFormat () {
                 getName() + "|" + getStartDate() + "|" + getEndDate() + "|" + getStartTime() + "|"
                 + getEndTime();
     }
+
+    public static VEvent eventToVEvent(Event event) {
+        java.util.Calendar startDate = new GregorianCalendar();
+        int intStartDate = parseInt(event.getStartDate().substring(2,5)) - 1;
+        System.out.println(intStartDate);
+        startDate.set(java.util.Calendar.MONTH, 3);
```
Comment (clarencepohh) [inline src/main/java/data/Event.java:107]: Are these commands abstractable? They look a little repetitive.

File: src/main/java/data/TaskManager.java [hunk 2]
```diff
@@ -305,44 +313,63 @@ public void addManager(Scanner scanner, WeekView weekView, MonthView monthView,
...
+        String startTime = startDateAndTime[1];
+
+        System.out.println("Enter the end date of this task, along with the end time separated by a space:");
+        String inputEndDateAndTime = scanner.nextLine().trim();
+        checkIfDateTimeInFormat(inputEndDateAndTime);
+        String[] endDateAndTime = inputEndDateAndTime.split(" ");
+        String endDate = endDateAndTime[0];
+        String endTime = endDateAndTime[1];
+
+        String [] startAndEndDates = new String[]{startDate, endDate};
```
Comment (clarencepohh) [inline src/main/java/data/TaskManager.java:355]: Good job adhering to naming convention for plurals!

File: src/main/java/data/TaskManager.java [hunk 3]
```diff
@@ -44,8 +49,8 @@ public class TaskManager {
      * @param taskType The TaskType of the task to be added.
      * @param dates A String array that contains the relevant dates for the task to be added.
      */
-    public static void addTask(LocalDate date, String taskDescription, TaskType taskType, String[] dates,
-                               String[] times)
+    public static void addTask(LocalDate date, String taskDescription, TaskType taskType,
+                               String[] dates, String[] times)
```
Comment (clarencepohh) [inline src/main/java/data/TaskManager.java:53]: JavaDocs not updated for new params

**Review by clarencepohh — COMMENTED** [review]
Thanks for creating an Exception package for greater clarity in our packages! We might want to do more testing for the new additions.

Also do check if all possible exception paths are checked for, or else they could lead to bugs being flagged in the PE.

File: src/main/java/data/exceptions/StorageFileException.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package data.exceptions;
```
Comment (Mahesh1772) [inline src/main/java/data/exceptions/StorageFileException.java:1]: Good job adding a class for Storing files exception

File: src/main/java/data/exceptions/TaskManagerException.java [hunk 5]
```diff
@@ -23,6 +24,29 @@ public TaskManagerException(String errorMessage) {
...
 
+    public static void checkIfDateTimeInFormat(String dateTime) throws TaskManagerException {
+        // Validate start date and time format
+        if (!dateTime.matches("\\d{2}/\\d{2}/\\d{4} \\d{4}")) {
+            throw new TaskManagerException("Invalid start date and time format. " +
+                    "Please use the format dd/MM/yyyy HHmm");
+        }
+    }
+
+    public static void checkIfDateInFormat(String date) throws TaskManagerException {
```
Comment (Mahesh1772) [inline src/main/java/data/exceptions/TaskManagerException.java:35]: Is it ok to just have exceptions for start date?

File: src/test/java/data/TaskManagerTest.java [hunk 6]
```diff
@@ -1,5 +1,6 @@
 package data;
 
+import data.exceptions.TaskManagerException;
```
Comment (Mahesh1772) [inline src/test/java/data/TaskManagerTest.java:3]: Amazing job adding a new package for all exceptions

**Review by Mahesh1772 — COMMENTED** [review]
Good job on using a new package and creating necessary Exception classes

**Pull request merged**